Jenkinsfile

```diff
@@ -1,15 +1,34 @@
 #!/usr/bin/groovy
 node{
   ws{
+
     stage 'checkout code'
-    checkout scm
+        checkout scm
+
+    stage 'build and unit test'
+    
+        kubernetes.pod('buildtestpod').withImage('maven')
+        .withPrivileged(true)
+        .withSecret('jenkins-maven-settings','/root/.m2')
+        .inside {
+            sh 'mvn -Dmaven.test.failure.ignore clean verify'
```

Comment:
Ignoring failures is required to capture test metrics per the [Jenkins workflow documentation](https://github.com/jenkinsci/workflow-plugin/blob/master/TUTORIAL.md)

The first phase only handles capturing unit test metrics and then the unit tests are run once again along with a full packaging and deployment